### DISCUSSION TITLE
Spaceship how to remove all territories?

### DISCUSSION THREAD

**Discussion by lawliet616** [fastlane/fastlane#19375] [Q&A]
This is my code:
```
  app = Spaceship::ConnectAPI::App.find(bundle_id)

  def remove_from_sale(app)
    attributes = {}
    attributes[:availableInNewTerritories] = false
    app.update(attributes: attributes, territory_ids: [])
  end
```

It updates the availableInNewTerritories setting correctly, but the territories remain unchanged.

If I use `territory_ids: ["ZWE"]`, it works, but naturally leaves a single territory checked.

I've checked the source code, and it doesn't try to update the territories, if the array is empty.
https://github.com/fastlane/fastlane/blob/e874a47c6e2e0e61590a03d3b71e75e5a505d1ce/spaceship/lib/spaceship/connect_api/tunes/tunes.rb#L135

How can I remove all territories?

**Comment by lucgrabowski** [comment]
I've opened a PR to allow for removing an app from sale: https://github.com/fastlane/fastlane/pull/20324
